### PR TITLE
Ankit | Stage | UI implementation 

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cv4gcnc

### CODE REVIEW

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 1]
```diff
@@ -1383,7 +1417,11 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
             };
             this.subscriptionId = subscription?.subscriptionId;
             let data = { ...request, ...this.subscriptionRequest };
-            this.componentStore.changePlan(data);
+            if(subscription?.duration === 'MONTHLY' && subscription?.region?.code !== 'GBR') {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]:
```suggestion
            if (subscription?.duration === 'MONTHLY' && subscription?.region?.code !== 'GBR') {
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]: Optional suggestion due to Stage  PR

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 2]
```diff
@@ -386,6 +388,44 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
...
+                                if (res.message) {
+                                    this.toasterService.showSnackBar('error', res.message);
+                                }
+                                return this.patchState({
+                                    razorpaySuccess: false
+                                });
+                            }
+                        },
+                        (error: any) => {
+                            this.toasterService.showSnackBar('error', 'Something went wrong! Please try again.');
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: use translation

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts]: Done

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]: Done

**Review by dvCodeWorld — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**